spec/datadog/appsec/contrib/active_record/mysql2_adapter_spec.rb

```diff
@@ -119,15 +119,33 @@
       User.find_by_sql("SELECT * FROM users WHERE name = 'Bob'").to_a
     end
 
-    it 'adds an event to processor context if waf result is a match' do
-      result = Datadog::AppSec::SecurityEngine::Result::Match.new(
-        events: [], actions: {}, derivatives: {}, timeout: false, duration_ns: 0, duration_ext_ns: 0
-      )
+    context 'when waf result is a match' do
+      let(:result) do
+        Datadog::AppSec::SecurityEngine::Result::Match.new(
+          events: [],
+          actions: { 'generate_stack' => { 'stack_id' => 'some-id' } },
+          derivatives: {},
+          timeout: false,
+          duration_ns: 0,
+          duration_ext_ns: 0
+        )
+      end
 
-      expect(Datadog::AppSec.active_context).to receive(:run_rasp).and_return(result)
-      expect(Datadog::AppSec.active_context.events).to receive(:<<).and_call_original
+      before do
+        allow(Datadog::AppSec.active_context).to receive(:run_rasp).and_return(result)
+      end
 
-      User.where(name: 'Bob').to_a
+      it 'adds an event to processor context if waf result is a match' do
+        expect(Datadog::AppSec.active_context.events).to receive(:<<).and_call_original
+
+        User.where(name: 'Bob').to_a
```

Comment:
If we do "call original" is there a point to put expectation like that? If we call original anyway you can control the list of events instead (also there is explicit event content check, so)

```suggestion
        expect { User.where(name: 'Bob').to_a }.to change(Datadog::AppSec.active_context.events).from([])

        
```

Reply:
thanks! Fixed in [873f03c](https://github.com/DataDog/dd-trace-rb/pull/4321/commits/873f03c6337f3dce37c52739574855df7b645ef1)